src/api/codewriter.rs

```diff
@@ -0,0 +1,187 @@
+use byteorder::{BigEndian, WriteBytesExt};
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::Error;
+use std::result::Result;
+
+use crate::objects::codeholder::CodeHolder;
+use crate::objects::constant::Constant;
+use crate::objects::instruction::Instruction;
+use crate::objects::register::{Register, RegisterLocation, RegisterReference};
+
+fn write_register(buf: &mut Vec<u8>, r: &Register) -> Result<(), Error> {
```

Comment:
This returns a result but never seems to return an Error

Reply:
The `write_u32` function returns a result containing an error, so the error is passed on for completeness.

Reply:
ok, I'll just mark this as resolved